### DISCUSSION TITLE
BPF filters for zeek/suricata not working

### DISCUSSION THREAD

**Discussion by tuamortemo** [Security-Onion-Solutions/securityonion#13862] [2.4]
### Version

2.4.110

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

16-32

### RAM

16-32

### Storage for /

200-300 GB

### Storage for /nsm

1 TB

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Relatively recently, I noticed that the BPF filters configured in config-suricata/zeek-(sensor field) do not work. There are changes in the "bpf" files in the /opt/so/conf directories on the sensors
I've need to filter traffic which use port 3260.
The filter I tried:
not port 3260
not dst port 3260
not (ip *.*.*.* and port 3260)
If I understand correctly, then all traffic associated with this port in the zeek and suricata modules should not be displayed in Hunt. But I still see suricata alerts and zeek datasets.

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
Just to be clear, you set your BPF in `SOC > Administration > Configuration > bpf`? This should work `not port 3260`,

**Reply by tuamortemo** [reply]
Yep. configured there
I have 4 sensors in use, all configured with bpf with subnet and port suppression. But only on one I keep seeing packets going through in Hunt. Today I tried to reinstall the sensor, but it did not give any result. The sensor has two interfaces, but as I understand it, bpf works on bond0 and that's not the reason. Could it be that more threads are needed for zeek and suricata, because there are only 12 CPUs on the sensor and 4 are allocated for each module (Inbound Monitor Traffic +-750Mb/s)?

**Reply by cm-ops** [reply]
What does your configuration look like?

**Reply by tuamortemo** [reply]
Manager, 3 search nodes, 5 forward nodes. 
2 Forward - 12 CPU, 32-64RAM
3 Forward  - 32+ CPU, 32-64RAM
Traffic monitoring via SPAN

**Reply by cm-ops** [reply]
I meant your BPF configuration in SOC, what does that look like?

**Reply by tuamortemo** [reply]
I apologize for the long wait
Zeek BPF config, which is used on all sensors and generally works when comparing tcpdump and soc
not dst portrange 2500-2525 &&
not port 902 &&
not port 3260

**Reply by cm-ops** [reply]
That BPF looks okay, so on that one sensor you are seeing traffic that is coming from one of those ports? But not on the other three?

**Reply by tuamortemo** [reply]
Yes, there are two more sensors with the same filters. Judging by tcpdump, there is also traffic with these ports, but they do not reach hunt

**Reply by cm-ops** [reply]
If they are reaching Hunt, they aren't making it to Elasticsearch, so I would think the BPF is working.
